### ISSUE TITLE
Reported maven plugin issues

### ISSUE THREAD

**Issue by lthoulon-locala** [deweyjose/graphqlcodegen#87]
Hello. Recently I get warning with some plugins. Here are those yours generate:

```
[WARNING]  * io.github.deweyjose:graphqlcodegen-maven-plugin:1.38
[WARNING]   Plugin issue(s):
[WARNING]    * Plugin is a Maven 2.x plugin, which will be not supported in Maven 4.x
[WARNING]    * Plugin mixes multiple Maven versions: [3.8.1, 2.2.1]
[WARNING]    * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-plugin-api:3.8.1, org.apache.maven:maven-artifact:3.8.1, org.apache.maven:maven-model:3.8.1]
[WARNING]    * Plugin depends on plexus-container-default, which is EOL
[WARNING] 
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
```

**Comment by deweyjose** [comment]
Hi @lthoulon-locala - what version of maven are you using?

**Comment by lthoulon-locala** [comment]
@deweyjose I'm using maven 3.9.2

```
me@mycomputer myproject % mvn --version
Apache Maven 3.9.2 (c9616018c7a021c1c39be70fb2843d6f5f9b8a1c)
Maven home: /opt/homebrew/Cellar/maven/3.9.2/libexec
Java version: 20.0.1, vendor: Homebrew, runtime: /opt/homebrew/Cellar/openjdk/20.0.1/libexec/openjdk.jdk/Contents/Home
Default locale: fr_FR, platform encoding: UTF-8
OS name: "mac os x", version: "13.4.1", arch: "aarch64", family: "mac"
```

**Comment by deweyjose** [comment]
@lthoulon-locala most of these are fixed in #115 which will be released as 1.51.1 tonight. I suggest upgrading to maven 3.9.6 as there are other core maven dependencies that I can't upgrade as a plugin builder.
